### PR TITLE
fix shuffle in target codes for skewed data

### PR DESCRIPTION
**Pull request by songhappy**
target_code has a big table join a relatively small table. the big table is always skewed because feature data is always long-tailed. Regular join will pull most of the data into certain nodes and fails; broadcast join will add more data into cluster, but the size is unknown. 
The solution here is, split the data into two parts, broadcast join the top 1m features,  and regular join the left over

### CODE REVIEW

File: python/friesian/src/bigdl/friesian/feature/utils.py [hunk 1]
```diff
@@ -160,13 +162,29 @@ def encode_target_(tbl, targets, target_cols=None, drop_cat=True, drop_fold=True
                     new_out_target_mean[out_col] = target_mean
             out_target_mean = new_out_target_mean
 
+        all_size = target_code.size()
+        limit_size = 1000000
```
Comment (hkvision) [inline python/friesian/src/bigdl/friesian/feature/utils.py:166]: how this limit size is determined?

**Comment by hkvision** [comment]
Please add some explanations for this PR and how the modification works.

File: python/friesian/src/bigdl/friesian/feature/utils.py (hunk 1 above)
Comment (songhappy) [reply python/friesian/src/bigdl/friesian/feature/utils.py:166]: most of the times, size of each categorical feature stay within 1m.

**Comment by songhappy** [comment]
Jenkins passed: 
http://10.112.231.51:18888/view/BigDL-2.0-PR/job/BigDL-PR-Friesian-Validation/198/

**Review by hkvision — APPROVED** [review]

**Pull request merged**